src/main/java/sb/service/OrderService.java

```diff
@@ -0,0 +1,182 @@
+package sb.service;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+import sb.domain.dto.request.OrderDTORequest;
+import sb.domain.entity.*;
+import sb.persistence.dao.*;
+import sb.service.exception.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class OrderService {
+
+    private OrderDAO orderDAO;
+    private UserDAO userDAO;
+    private ProductDAO productDAO;
+    private StockDAO stockDAO;
+    private LineItemDAO lineItemDAO;
+
+
+    public OrderService(OrderDAO orderDAO, UserDAO userDAO, ProductDAO productDAO, StockDAO stockDAO, LineItemDAO lineItemDAO) {
+        this.orderDAO = orderDAO;
+        this.userDAO = userDAO;
+        this.productDAO = productDAO;
+        this.stockDAO = stockDAO;
+        this.lineItemDAO = lineItemDAO;
+    }
+
+
+    public List<Order> getAll() {
+        SecurityContext context = SecurityContextHolder.getContext();
+        User user = userDAO.getByName(context.getAuthentication().getName())
+                .orElseThrow(() -> new NotFoundException("User is not found"));
+
+        if(user.getRole().equals("admin")) {
+            return orderDAO.getAll();
+        } else {
+            return orderDAO.getByUser(user.getId());
+        }
+    }
+
+
+    public Order get(int id) {
+        return orderDAO.get(id).orElseThrow(
+                () -> new NotFoundException("No such order")
+        );
+    }
+
+    public List<Order> getByUser(int id) {
+        return orderDAO.getByUser(id);
+    }
+
+
+    @Transactional
+    public Order create(OrderDTORequest orderDTORequest) {
+        SecurityContext context = SecurityContextHolder.getContext();
+        User user = userDAO.getByName(context.getAuthentication().getName())
+                .orElseThrow(() -> new NotFoundException("No such user")
+        );
+
+        List<LineItem> lineItems = orderDTORequest.getLineItems();
+
+        Optional<Product> product;
+        Optional<Stock> stock;
```

Comment:
It seems like these fields are not used.

Reply:
fix: remove